Add new front matter element: excerpt

kathryngraysonnanz opened the pull request:
This commit adds the excerpt front matter element, to show an abbreviated version of the definition in social previews. This also includes the documentation updates to support this change.

Commit:
This commit adds the excerpt front matter element, to show an
abbreviated version of the definition in social previews. This
also includes the documentation updates to support this change.

kathryngraysonnanz commented:
This PR addresses the following issues: 
https://github.com/selfdefined/web-app/issues/179
https://github.com/selfdefined/web-app/issues/225

ovlb commented on 11ty/_includes/layouts/base.njk:
```diff
@@ -6,10 +6,24 @@
     <meta
       name="description"
-      content="{{ renderData.description or description or metadata.description }}"
+      content="{{ alert + ": " + preview }}"
```
We can `{% set %}` this combination at the beginning of the file to avoid having to repeat this in `og:description`, `twitter:description` and `description`.

ovlb reviewed (commented):
There’s one small optimisation that came to mind, other than that I’d say this PR is good to go. 🐚

tatianamac commented:
(Thank you, @kathryngraysonnanz and @ovlb !)

tatianamac reviewed (approved):
Adding an approving review based on @ovlb 's last note so that this can be merged in.

Commit:
This commit adds @olvb's suggestion to use 'set' to define the
alert & preview combo, to avoid repetitive code

kathryngraysonnanz commented:
@tatianamac @ovlb Thanks for the feedback! I've incorporated it, so we should be good to go.

ovlb committed:
Merge branch 'prod' into issue/add_excerpt

ovlb commented:
I tested this in Slack with myself quickly and I’m wondering if the «warning» makes sense in the isolation of a share preview? Avoid is clear enough, but warning kind of vague. What do you think, @tatianamac, @kathryngraysonnanz?

<img width="625" alt="Screenshot of the sharing preview for ableism and barbaric" src="https://user-images.githubusercontent.com/7645967/85209044-209cb180-b335-11ea-8d0d-7349e61352e9.png">

(Semi-unrelated: Shall we create a follow-up-issue to design a generic share picture to liven the preview up a bit?)

tatianamac commented:
> I tested this in Slack with myself quickly and I’m wondering if the «warning» makes sense in the isolation of a share preview? Avoid is clear enough, but warning kind of vague. What do you think, @tatianamac, @kathryngraysonnanz?
> 
> <img alt="Screenshot of the sharing preview for ableism and barbaric" width="625" src="https://user-images.githubusercontent.com/7645967/85209044-209cb180-b335-11ea-8d0d-7349e61352e9.png">
> 
> (Semi-unrelated: Shall we create a follow-up-issue to design a generic share picture to liven the preview up a bit?)

I agree. I don't think warning makes sense in this context; we should suppress content warning in the preview as it requires the context of the box.

kathryngraysonnanz commented:
So, do we feel like the right move on this is to just remove the flag content in the preview? I'm good with that, just wanted to confirm before making changes. 

Also @ovlb – I think the idea of a generic image for social is a great one. Good call.

tatianamac commented:
I think we should remove flag content for content warnings, but not for avoid.

ovlb commented:
Agreed. Avoid makes sense in the context of the preview.

ovlb committed:
Merge branch 'prod' into issue/add_excerpt

ovlb commented:
@kathryngraysonnanz Do you have bandwidth to incorporate the removal of the warning flag to finish off this PR? 

O*

kathryngraysonnanz commented:
@ovlb yes, sorry – started a new job recently and life has been hectic, haha. i'll try to knock this out tonight

Commit:
This commit updates the preview text to only include the flag if
it's "avoid" – if the flag is "warning" or there is no flag, it
will not appear at the beginning of the preview text.

ajtran303 commented:
> This branch is out-of-date with the base branch

Is this a good spot in the codebase / workflow for me to hop in?

I found the [documentation for setting up my development environment](https://www.selfdefined.app/documentation/development/) . 

Is the next step here to get the PR all set to merge by resolving the 'out-of-date' check?

ovlb commented:
Hey @ajtran303!

> Is this a good spot in the codebase / workflow for me to hop in?

Unfortunately, it’s a rather complex spot to get going. As Kathryn has been working on their fork, you’d need to fork their repo, open a PR over there, which Kathryn needs to merge in so the change appears here. 

But, if you’d like a small issue/PR to get going, we’ve one action item left here, which is the addition of a generic preview image via `og:image` and `twitter:image`. Do you feel comfortable working on this? This does not necessarily include creating the image, if you aren’t comfortable with graphic design, but rather adding the code.

ovlb committed:
Merge branch 'prod' into issue/add_excerpt

ajtran303 commented:
> But, if you’d like a small issue/PR to get going, we’ve one action item left here, which is the addition of a generic preview image via `og:image` and `twitter:image`. Do you feel comfortable working on this? This does not necessarily include creating the image, if you aren’t comfortable with graphic design, but rather adding the code.

Cool! How is this for a PR name? `Add new front matter element: image preview`

ovlb commented:
@kathryngraysonnanz Thanks for your work on this. And forgive my sloppy reviewing. As it’s ready to merge now, would you like us to thank you personally when we announce the feature on Twitter?

@ajtran303 That’s an on point name :)
